src/metal.cs

```diff
@@ -1741,6 +2001,69 @@ partial interface MTLRenderCommandEncoder : MTLCommandEncoder {
 #endif
 		[Export ("drawIndexedPatches:patchIndexBuffer:patchIndexBufferOffset:controlPointIndexBuffer:controlPointIndexBufferOffset:indirectBuffer:indirectBufferOffset:")]
 		void DrawIndexedPatches (nuint numberOfPatchControlPoints, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, IMTLBuffer controlPointIndexBuffer, nuint controlPointIndexBufferOffset, IMTLBuffer indirectBuffer, nuint indirectBufferOffset);
+
+		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setViewports:count:")]
+		void SetViewports (IntPtr viewports, nuint count);
```

Comment:
The generator doesn't appear to support doing this.  When I do that for the binding, it gives a build error that: 
```
/Users/Tim/Projects/Xamarin/xamarin-macios-xcode9/xamarin-macios/src/build/mac/mobile/Metal/MTLRenderCommandEncoder.g.cs(530,47): error CS1503: Argument `#1' cannot convert `Metal.MTLViewport[]' expression to type `Foundation.NSObject[]'
```

MTLViewport is a struc, not an NSObject, and it looks like it's trying to convert it to an NSArray of NSObjects.  The original objective-c is:

```
- (void)setViewports:(const MTLViewport *)viewports 
               count:(NSUInteger)count;
```